### PR TITLE
Override dict's get method

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Merges #474 and tests.

## Approach

One sentence in the pydoc was a little bit misleading, so I changed it to:
> Dictates whether a remote request is made to retrieve the doc, if it is not present in the local cache.

Extended the tests to cover the new behaviour based on the proposed behaviour table.

### Proposed behaviour

| Call | Doc is local | Doc is remote | Result | Covered with tests |
| --- | --- | --- | --- | --- |
| `db['foo']` | `True` | `True` | returns local doc | test_create_document_that_already_exists<br>test_create_document_without_id<br>test_create_design_document<br>test_create_empty_document
| `db['foo']` | `False` | `True` | returns remote doc | test_get_document_from_remote |
| `db['foo']` | `False` | `False` | returns `KeyError` | test_get_non_existing_doc_via_getitem
| `db['foo']` | `True` | `False` | returns local doc | test_get_document_from_cache |
| `db.get('foo')` | `True` | `True` | returns local doc | test_create_document_that_already_exists<br>test_create_document_without_id<br>test_create_design_document<br>
| `db.get('foo')` | `False` | `True` | returns `None` | test_get_document_from_remote |
| `db.get('foo')` | `False` | `False` | returns `None` | test_get_non_existing_document_from_cache |
| `db.get('foo')` | `True` | `False` | returns local doc | test_get_document_from_cache |
| `db.get('foo', remote=True)` | `True` | `True` | returns local doc | test_create_document_that_already_exists<br>test_create_document_without_id<br>test_create_design_document<br>test_create_empty_document
| `db.get('foo', remote=True)` | `False` | `True` | fetches remote doc | test_get_document_from_remote |
| `db.get('foo', remote=True)` | `False` | `False` | returns `None` | test_get_non_existing_document_from_remote |
| `db.get('foo', remote=True)` | `True` | `False` | returns local doc | test_get_document_from_cache |

## Testing

New assertions to cover the new feature to:
* `test_create_document_with_id`
* `test_create_document_that_already_exists`
* `test_create_document_without_id`
* `test_create_design_document`
* `test_create_empty_document`

New test cases:
* `test_get_non_existing_document_from_remote`
* `test_get_non_existing_document_from_cache`
* `test_get_document_from_cache`
* `test_get_document_from_remote`

## Monitoring and Logging

- No change

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 - [FIXED] Incorrect use of username as account name in `Cloudant.bluemix()`.
 - [IMPROVED] Documented use of None account name and url override for `Cloudant.iam()`.
 - [FIXED] Use custom encoder (if provided) for all view `key` params not just `keys`.
+- [NEW] Override the `get()` method for CouchDatabase, allowing it to retrive remote docuemnts if the `remote` parameter is specified.
```
Comment (ricellis) [inline CHANGES.md:9]:
```suggestion
- [NEW] Override `dict.get` method for `CouchDatabase` to add `remote` parameter allowing it to retrieve a remote document if specified.
```

File: tests/unit/database_tests.py [hunk 2]
```diff
@@ -271,13 +273,51 @@ def test_create_document_with_id(self):
...
+        self.assertIsNone(doc)
+
+    def test_get_document_from_cache(self):
+        """
+        Test dict's get on a document from cache.
+        """
+        doc = Document(self.db, document_id='julia06')
+        self.db['julia06'] = doc
+        self.assertEqual(self.db.get('julia06'), doc)
+        # fetch the local doc independently from the value of the remote param
```
Comment (ricellis) [inline tests/unit/database_tests.py:297]:
```suggestion
        # doc is fetched from the local dict preferentially to remote even with remote=True
```

**Review by ricellis — APPROVED** [review]
LGTM, couple of suggestions for comment improvements

**Review by emlaver — APPROVED** [review]

**Comment by mojito317** [comment]
Thank you @dominickj-tdi for the implementation. Thanks, @ricellis, and @emlaver for the reviews. I'm about to merge this.

**Pull request merged**